src/CoreShop/Bundle/FrontendBundle/Controller/ProductController.php

```diff
@@ -47,6 +47,10 @@ public function detailAction(Request $request)
         if (!$product->isPublished() || $product->getActive() !== true) {
             throw new NotFoundHttpException('product not found');
         }
+        
+        if(!in_array($this->get('coreshop.context.store')->getStore()->getId(),$product->getStores())){
```

Comment:
looks good, except for your code formatting. Can you please fix that? (What I mean is the missing whitespace after the comma)